src/SignalR/clients/csharp/Client/src/HttpConnectionFactory.cs

```diff
@@ -36,17 +45,46 @@ public HttpConnectionFactory(IOptions<HttpConnectionOptions> options, ILoggerFac
                 throw new ArgumentNullException(nameof(loggerFactory));
             }
 
+            _transferFormat = hubProtocol.TransferFormat;
             _httpConnectionOptions = options.Value;
             _loggerFactory = loggerFactory;
         }
 
-        /// <inheritdoc />
-        public async Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat, CancellationToken cancellationToken = default)
+        /// <summary>
+        /// Creates a new connection to an <see cref="HttpEndPoint"/>.
+        /// </summary>
+        /// <param name="httpEndPoint">The <see cref="HttpEndPoint"/> to connect to.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
+        /// <returns>
+        /// A <see cref="ValueTask{TResult}"/> that represents the asynchronous connect.
+        /// The <see cref="Task{TResult}.Result"/> property returns a <see cref="ConnectionContext"/> for the new connection.
+        /// </returns>
+        public async ValueTask<ConnectionContext> ConnectAsync(EndPoint httpEndPoint, CancellationToken cancellationToken = default)
```

Comment:
Rename httpEndPoint to endPoint

Reply:
This is the HttpConnectionFactory that you want to make internal, so does it matter? Only HttpEndPoints will work anyway.

Reply:
So you don't have to have a strange name like "castedHttpEndpoint".